### PR TITLE
Add case sensitivity in search and replace tool.

### LINKED ISSUE #25: Add case sensitivity in "find and replace"

### PR DESCRIPTION
**Pull request by fede-basseni**
I added case-sensitivity checkbox in search and replace tool.

### CODE REVIEW

File: src/src/language.cpp [hunk 1]
```diff
@@ -86,12 +86,14 @@ const std::vector<Key> &Language::getKeys()
...
-void Language::replaceMessage(const std::string &oldMsg, const std::string &newMsg)
+void Language::replaceMessage(const std::string &oldMsg, const std::string &newMsg,
+                              const int caseSensitivity)
 {
     for (const auto &key : keys) {
         // Fuck off I'm using QStrings...
         auto msgAsQString = QString::fromStdString(messages.at(key));
-        msgAsQString.replace(oldMsg.c_str(), newMsg.c_str());
+        msgAsQString.replace(oldMsg.c_str(), newMsg.c_str(),
+                            (caseSensitivity) ? Qt::CaseSensitive : Qt::CaseInsensitive);
```
Comment (guerinoni) [inline src/src/language.cpp]: remove parenthesis

File: src/src/mainwindow.cpp [hunk 2]
```diff
@@ -475,19 +475,31 @@ void MainWindow::findAndReplace()
 {
     QDialog d(this);
     d.setLayout(new QVBoxLayout);
+    auto sLabel = new QLabel(&d);
+    sLabel->setText("Search:");
```
Comment (guerinoni) [inline src/src/mainwindow.cpp]: make string translatable

File: src/src/mainwindow.cpp [hunk 3]
```diff
@@ -475,19 +475,31 @@ void MainWindow::findAndReplace()
...
     auto findTextEdit = new QLineEdit(&d);
+    auto rLabel = new QLabel(&d);
+    rLabel->setText("Replace with:");
```
Comment (guerinoni) [inline src/src/mainwindow.cpp]: make string translatable

File: src/src/mainwindow.cpp [hunk 4]
```diff
@@ -475,19 +475,31 @@ void MainWindow::findAndReplace()
...
     auto replaceTextEdit = new QLineEdit(&d);
-    d.layout()->addWidget(findTextEdit);
-    d.layout()->addWidget(replaceTextEdit);
+    auto caseSensitiveOption = new QCheckBox(&d);
+    caseSensitiveOption->setText("Case sentive");
```
Comment (guerinoni) [inline src/src/mainwindow.cpp]: make string translatable

**Review by guerinoni — CHANGES_REQUESTED** [review]

File: src/src/language.cpp [hunk 5]
```diff
@@ -86,12 +86,14 @@ const std::vector<Key> &Language::getKeys()
...
-void Language::replaceMessage(const std::string &oldMsg, const std::string &newMsg)
+void Language::replaceMessage(const std::string &oldMsg, const std::string &newMsg,
+                              int caseSensitivity)
 {
     for (const auto &key : keys) {
         // Fuck off I'm using QStrings...
         auto msgAsQString = QString::fromStdString(messages.at(key));
-        msgAsQString.replace(oldMsg.c_str(), newMsg.c_str());
+        msgAsQString.replace(oldMsg.c_str(), newMsg.c_str(),
+                            caseSensitivity ? Qt::CaseSensitive : Qt::CaseInsensitive);
```
Comment (guerinoni) [inline src/src/language.cpp:96]: wrong indentation

**Comment by guerinoni** [comment]
I tried this with a string "lol" and "LOL", replacing "lol" with case sensitive flag changes all item

**Comment by fede-basseni** [comment]
> I tried this with a string "lol" and "LOL", replacing "lol" with case sensitive flag changes all item

I retried the function with case sensitive but I didn't see any problems. 
It's working correctly.

**Comment by guerinoni** [comment]
> > I tried this with a string "lol" and "LOL", replacing "lol" with case sensitive flag changes all item
> 
> I retried the function with case sensitive but I didn't see any problems.
> It's working correctly.

it works, but there are some issues editing a value in the table... because of this I reported some issues on this feature

**Review by guerinoni — APPROVED** [review]

**Review by AndreaRicchi — APPROVED** [review]
LGTM!

**Pull request merged**